addon/components/file-dropzone.ts

```diff
@@ -0,0 +1,354 @@
+    if (this.dataTransferWrapper) {
+      this.dataTransferWrapper.dataTransfer = event.dataTransfer;
+    }
+
+    if (!this.isAllowed) {
+      event.dataTransfer.dropEffect = this.cursor;
+      this.dataTransferWrapper = undefined;
+      return;
+    }
+
+    // @TODO - add tests for these or remove them
+    // // Testing support for dragging and dropping images
+    // // from other browser windows
+    // let url;
+
+    // // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+    // // @ts-ignore
+    // const html = this.dataTransferWrapper.getData('text/html');
+    // if (html) {
+    //   const parsedHtml = parseHTML(html);
+    //   // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+    //   // @ts-ignore
+    //   const img = parsedHtml.getElementsByTagName('img')[0];
+    //   if (img) {
+    //     url = img.src;
+    //   }
+    // }
+
+    // if (url == null) {
+    //   // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+    //   // @ts-ignore
+    //   url = this.dataTransferWrapper.getData('text/uri-list');
+    // }
+
+    // if (url) {
+    //   const image = new Image();
+    //   const [filename] = url.split('/').slice(-1);
+    //   image.crossOrigin = 'anonymous';
+    //   image.onload = () => {
+    //     const canvas = document.createElement('canvas');
+    //     canvas.width = image.width;
+    //     canvas.height = image.height;
+
+    //     const ctx = canvas.getContext('2d');
+    //     ctx?.drawImage(image, 0, 0);
+
+    //     if (canvas.toBlob) {
+    //       canvas.toBlob((blob) => {
+    //         // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+    //         // @ts-ignore
+    //         const [file] = this.addFiles([blob], FileSource.web);
+    //         // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+    //         // @ts-ignore
+    //         file.name = filename;
+    //       });
+    //     } else {
+    //       const binStr = atob(canvas.toDataURL().split(',')[1]);
+    //       const len = binStr.length;
+    //       const arr = new Uint8Array(len);
+
+    //       for (let i = 0; i < len; i++) {
+    //         arr[i] = binStr.charCodeAt(i);
+    //       }
+    //       const blob = new Blob([arr], { type: 'image/png' });
+    //       // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+    //       // @ts-ignore
+    //       blob.name = filename;
+    //       // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+    //       // @ts-ignore
+    //       const [file] = this.addFiles([blob], FileSource.web);
+    //       // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+    //       // @ts-ignore
+    //       file.name = filename;
+    //     }
+    //   };
+    //   /* eslint-disable no-console */
+    //   image.onerror = function (e) {
+    //     console.log(e);
+    //   };
+    //   /* eslint-enable no-console */
+    //   image.src = url;
+    // }
```

Comment:
This code concerns me as I have found there is no test coverage of it and I'm not sure how to test it manually.

I'm worried we may have broken this functionality without realising already. However I did try to drag an image from another browser window in `v4.0.3` and it didn't work so 🤷🏻‍♂️ .

Options:
- Explicitly remove this as it was undocumented and untested. Can always be added back as an on-breaking enhancement
- Try and figure out how it works and support it now